pkg/pipeline/decode/decode_protobuf.go

```diff
@@ -0,0 +1,85 @@
+package decode
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/netobserv-agent/pkg/pbflow"
+	"github.com/sirupsen/logrus"
+)
+
+var pflog = logrus.WithField("component", "Protobuf")
+
+// Protobuf decodes protobuf flow records definitions, as forwarded by
+// ingest.NetObservAgent, into a Generic Map that follows the same naming conventions
+// as the IPFIX flows from ingest.IngestCollector
+type Protobuf struct {
+}
+
+func NewProtobuf() (Decoder, error) {
+	return &Protobuf{}, nil
+}
+
+// Decode decodes input strings to a list of flow entries
+func (c *Protobuf) Decode(in []interface{}) []config.GenericMap {
+	if len(in) == 0 {
+		pflog.Warn("empty input. Skipping")
+		return []config.GenericMap{}
+	}
+	pb, ok := in[0].(*pbflow.Records)
+	if !ok {
+		pflog.WithField("type", fmt.Sprintf("%T", pb)).
+			Warn("expecting input to be *pbflow.Records. Skipping")
+	}
+	out := make([]config.GenericMap, 0, len(pb.Entries))
+	for _, entry := range pb.Entries {
+		out = append(out, pbFlowToMap(entry))
+	}
+	return out
+}
+
+func pbFlowToMap(flow *pbflow.Record) config.GenericMap {
+	if flow == nil {
+		return config.GenericMap{}
+	}
+	out := config.GenericMap{
```

Comment:
Should this be built somehow dynamic from https://github.com/netobserv/netobserv-agent/blob/main/proto/flow.proto ????

Reply:
That would involve flattening/changing the `flow.proto` definition to match the map structure, and then create some sort of code generation or use reflection to create the map. Since the number of fields is still relatively small, I'd keep using this explicit conversion, and consider another approach in the future if we feel that the fields grow until making this conversion unmaintainable.

Reply:
@mariomac this is exactly my fear :-( .... that this will grow and no one will be able to find where this came from and why this is not working. Any chance to someone connect that to the source (i.e. to the https://github.com/netobserv/netobserv-agent) project ... if not in code than at least with some remarks maybe ?

Reply:
Yeah, actually the `*pbflow.Records` type is imported from the `github.com/netobserv/netobserv-agent` directly, so IDE navigation leads you to the original protobuf definition (at least in Goland IDE).

But I agree that we should investigate a more automatic way to map protobuf --> generic map fields that does not penalize readability nor performance.